src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -41,10 +41,8 @@
 use function is_string;
 use function strtolower;
 
-abstract class ReflectionFunctionAbstract
+trait ReflectionFunctionAbstract
```

Comment:
Can we enforce implementations here, at type level?

Reply:
Added two more commits.

Reply:
I was more wondering if we had a way to use `@psalm-require-extends A|B` or such :thinking:

Reply:
I tried to add 

```
/**
 * @psalm-require-extends ReflectionMethod|ReflectionFunction
 */
 ```
 
 and Psalm does not report anything. However Psalm does not report anything even without the annotation... Should I add the annotation?

Reply:
No, my bad... In this branch it reports this:

```
ERROR: ExtensionRequirementViolation - src/Reflection/ReflectionFunction.php:25:9 - Roave\BetterReflection\Reflection\ReflectionFunctionAbstract requires using class to extend Roave\BetterReflection\Reflection\ReflectionFunction|Roave\BetterReflection\Reflection\ReflectionMethod, but Roave\BetterReflection\Reflection\ReflectionFunction does not (see https://psalm.dev/239)
    use ReflectionFunctionAbstract;


ERROR: ExtensionRequirementViolation - src/Reflection/ReflectionMethod.php:29:9 - Roave\BetterReflection\Reflection\ReflectionFunctionAbstract requires using class to extend Roave\BetterReflection\Reflection\ReflectionFunction|Roave\BetterReflection\Reflection\ReflectionMethod, but Roave\BetterReflection\Reflection\ReflectionMethod does not (see https://psalm.dev/239)
    use ReflectionFunctionAbstract;
```

Reply:
Right, let's roll with it for now - we may remove this trait in a much further placed release, if needed :)